Shopify/ruby-lsp
# Support DocumentLink request

### Motivation

This request uses the gem source comments generated by Tapioca to let users quickly jump to the actual source of the definition.

### Implementation

Add a new `DocumentLink` request to detect `# source://PATH_TO_FILE:line` comments and return clickable links if the file exists.

### Automated Tests

New expectation + integration test cases.

### Manual Tests

1. Install the `vscode-ruby-lsp` extension from the [PR](https://github.com/Shopify/vscode-ruby-lsp/pull/170)
2. Check out to the branch and restart LSP
3. Run `bundle exec tapioca gem` to make sure we have rbi files that have the new source comments
4. Find a file that has `# source://PATH_TO_FILE:line` comments and hover the mouse over one of them
    - It should show `Jump to PATH_TO_FILE#line` in the tooltip
    - If you click it, it should take it to the exact location


**Demo**

![document_link](https://user-images.githubusercontent.com/5079556/181105085-b371d81e-280f-4871-ac07-67b190f59571.gif)
